existential-vs-universal-quantification/article.md

```diff
@@ -0,0 +1,264 @@
+# Existential and Universal Quantification in Haskell
+
+This article gives a brief introduction into one of the most common Haskell extensions – `ExplicitForAll` – and syntax constructions that this extension brings to language.
+I haven't yet found any big project that wouldn't use this extension, so it is very
+useful to either become acquainted with it or to refresh it in your memory if you already know about it.
+
+Stay tuned if you want to know how to create lists of differently typed variables, how to add complex constraints, and more!
+
+
+## Introduction
+
+From predicate logic, we know about the existence of two quantifiers – `$\forall$` and `$\exists$` – that we can apply to a term `x` and some statement.
+The former means: "For every term x, the statement is true", the later: `"There is at least one term x for which this statement is true"`.
+
+Since Haskell is based on predicate logic, there is a place for these quantifiers in language syntax. However, in Haskell, we are talking not about some terms and statements but about types.
+
+```haskell
+id :: forall a. a -> a
+id x = x
+```
+
+This quantification is implicit by default (if you check the type signature of `id`, there is no quantifier there). To make it explicit, you need to use a specific extension – "ExplicitForAll".
```

Comment:
How would you mantion HMTI here, after showing true `id` type? I thought about something like
"This quantifiers come from Hindley–Milner type system ...", but I don't know what to add instead of ...